lib/membrane_rtc_engine/endpoints/hls/mobile_layout_maker.ex

```diff
@@ -7,21 +7,34 @@ defmodule Membrane.RTC.Engine.Endpoint.HLS.MobileLayoutMaker do
   require Membrane.Pad
 
   alias Membrane.Pad
-  alias Membrane.VideoCompositor.RustStructs.VideoPlacement
+  alias Membrane.VideoCompositor.RustStructs.BaseVideoPlacement
+  alias Membrane.VideoCompositor.VideoTransformations
+  alias Membrane.VideoCompositor.VideoTransformations.TextureTransformations.CornersRounding
+  alias Membrane.VideoCompositor.VideoTransformations.TextureTransformations.Cropping
 
   @impl true
   def init(output_caps), do: %{tracks: %{}, output_caps: output_caps}
+
   @impl true
-  def track_added(state, %{metadata: %{"mainPresenter" => true}} = track) do
-    new_state = put_in(state, [:tracks, track.id], track)
-    updated_layout = [{Pad.ref(:input, track.id), get_track_layout(:main, 0, state.output_caps)}]
+  def track_added(state, %{metadata: %{"mainPresenter" => true}} = track, caps) do
+    new_state = put_in(state, [:tracks, track.id], {track, caps})
+
+    {trans, layout} = get_track_layout(:main, 0, caps, state.output_caps)
```

Comment:
0 is a little bit of a magic number, what do you think about moving it to a constant?